src/Microsoft.Android.Sdk.Analysis/Analyzers/CustomApplicationAnalyzer.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.Android.Sdk.Analysis;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+[DiagnosticAnalyzer (LanguageNames.CSharp)]
+public class CustomApplicationAnalyzer : DiagnosticAnalyzer
+{
+	private const string AndroidApplication = "Android.App.Application";
+	public const string DiagnosticId = "DNAA0001";
+	private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor (
+		id: DiagnosticId,
+		title: Resources.DNAA0001_Title,
+		messageFormat: Resources.DNAA0001_MessageFormat,
+		category: "Usage",
+		defaultSeverity: DiagnosticSeverity.Warning,
+		isEnabledByDefault: true
+	);
+
+	public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create (Rule);
+
+	public override void Initialize (AnalysisContext context)
+	{
+		context.ConfigureGeneratedCodeAnalysis (GeneratedCodeAnalysisFlags.None);
+		context.EnableConcurrentExecution ();
+
+		// Register a syntax node action to analyze method declarations
+		context.RegisterSyntaxNodeAction (AnalyzeClass, SyntaxKind.ClassDeclaration);
```

Comment:
Is there an `AnalyzeConstructor` option for this callback? I wonder if that will be faster, because the IDE runs these analyzers as you type. I used a Property one here:

* https://github.com/jonathanpeppers/memory-analyzers/blob/2bc83bcb6ab0fd4673a646f1b7bbb2e574cdeac5/MemoryAnalyzers/MemoryAnalyzers/MemoryAnalyzer.cs#L95

Reply:
Oh, I guess class makes sense, as you need it to run if there _is no_ constructor.

Reply:
Ok, yeah I think you found it, it's on the Symbol:

![image](https://github.com/user-attachments/assets/4fad8803-1140-4ec0-a3fe-13dcc72d44dc)

^^ copilot typed most of that